File: LibraryApp/Application/Loans/DTOs/LoanDto.cs
namespace Application.Loans.DTOs;

public class LoanDto
{
    public int Id { get; set; }
    public DateTime LoanDate { get; set; }
    public DateTime DueDate { get; set; }
        
    // İade edilmeme durumuna karşı nullable (?) yapıldı
    public DateTime? ReturnDate { get; set; } 
        
    // AutoMapper ile Member ve Book tablolarından çekilecek alanlar
    public string BookName { get; set; } 
    public string FullName { get; set; }
}

File: angular/src/app/features/loans/models/loan.dto.ts
export interface LoanDto {
  id: number;
  loanDate: string | Date;
  dueDate: string | Date;
  returnDate?: string | Date | null;
  bookName: string;
  memberFullName: string;
}

export interface LoanCreateDto {
  memberID: number;
  bookID: number;
  loanDate: string | Date;
  dueDate: string | Date;
  returnDate?: string | Date | null;
}